### ISSUE TITLE
Famous Question Badge

### ISSUE THREAD

**Issue by jophab** [jophab/StackOverflow--Badge-Hunter#2]
Famous Question Badge is awarded to Questions with 10,000 views.

These are some meta.stackexchange posts that tells about how question views work

- http://meta.stackexchange.com/questions/36728/how-are-the-number-of-views-in-a-question-calculated

- http://meta.stackexchange.com/questions/87092/dissecting-the-stack-overflow-views-counter

- http://meta.stackoverflow.com/questions/262802/can-viewing-your-own-question-cause-its-view-count-to-increase

we need to change ip address on each request, or schedule some script to visit the question link in regular intervals.
 
Is anybody having an idea to do this?
@ParadoxZero @gautamkrishnar

**Comment by gautamkrishnar** [comment]
i will give it a try...

**Comment by ParadoxZero** [comment]
Seems like the 15-minute timer will work. Also, we could get a centralized server, then we could implement some sort of peer view increase as in, every 15 minutes it will fetch the URLs it needs to visit and visit them if you want your URL to be visited then send it to the server. If we can have a **server** to do that, then rest is easy.

**edit**: We can use **_hostinger_**  or similar service and write a simple PHP code to hand out a JSON containing URLs. Then another links to add and delete URLs

**Comment by gautamkrishnar** [comment]
The problem is that the SO views will increase only if it is visited from different IP. I tried implementing this by using proxy and python request. But i dont know why, it failed. The python requests is always receiving a **200**. I think it is the problem with the proxy server i used. I don't know any proxy providers who provides free and unlimited service.

**Comment by ParadoxZero** [comment]
That's why I said having a **peer** approach will work, When the cache expires by yourself you can only add 1 count. But imagine it 20 people start using it? what if more than 20? 30? 50? 

If a different PC altogether visits it. Then your count will definitely increase without needing to change IP

**Comment by jophab** [comment]
I think online proxies are working.

There are so many online proxy services available.  Visiting the question  through them can increase count.

just google search "free online proxy".
Try them with a question url

@ParadoxZero Congo on 300 reps :wink:

**Comment by ParadoxZero** [comment]
So I guess in that case build a list of online proxies which can be manipulated using URLs.

PS: That was a huge jump 300 reps did you go and upvote all my answers, saw many upvotes pop up suddenly xD

**Comment by jophab** [comment]
I up voted some. Then u reached 200 reps. At 200 rep u got site  association bonus 100 reps. Thus 300 :smile: 

> So I guess in that case build a list of online proxies which can be manipulated using URLs.
 
I think so. What do you say? @ParadoxZero @gautamkrishnar 
Any better ideas?

**Comment by ParadoxZero** [comment]
It's either that or a peer-based approach as far as I can see. Otherwise, time visit should be possible but too slow.

**Comment by gautamkrishnar** [comment]
Please see the initial version. We just need to add the proxy provider's authentication URL.

**Comment by jophab** [comment]
We need many proxies right?

**Comment by gautamkrishnar** [comment]
Yes we need a proxy list, because response time of each proxies are different. I would recommend a threaded implementation that chooses each proxy from the list and execute requests on each proxy independently...

**Comment by ParadoxZero** [comment]
>  I would recommend a threaded implementation that chooses each proxy from the list and execute requests on each proxy independently

Threading would be an overkill since we don't need it to be interactive. It can take as long as it needs to right? I mean what's the hurry...

**Comment by gautamkrishnar** [comment]
@ParadoxZero 👍

**Comment by jophab** [comment]
@ParadoxZero Can you complete the peer based approach to get rid of this badge? :wink:

**Comment by ParadoxZero** [comment]
It will take a bit of time as I am a bit busy with a few of my projects.

**Comment by jophab** [comment]
@ParadoxZero Yeah Its fine :+1:
